launcher/ui/pages/instance/ManagedPackPage.cpp

```diff
@@ -41,8 +41,38 @@ class NoBigComboBoxStyle : public QProxyStyle {
         return QProxyStyle::styleHint(hint, option, widget, returnData);
     }
     // clang-format on
+
+    static NoBigComboBoxStyle* GetInstance(QStyle* style);
+
+   private:
+    static QMap<QStyle*, NoBigComboBoxStyle*> s_singleton_instances_;
+    static std::mutex s_singleton_instances_mutex_;
 };
 
+QMap<QStyle*, NoBigComboBoxStyle*>  NoBigComboBoxStyle::s_singleton_instances_ = {};
+std::mutex NoBigComboBoxStyle::s_singleton_instances_mutex_;
+
+/**
+ * QProxyStyle and QStyle objects object to being freed even if all the widgets using them are gone
+ * so make singlestons tied to the lifetime of the application to clean them up and ensure they arn't 
+ * being remade over and over agian leaking memory.
+ * */
+NoBigComboBoxStyle* NoBigComboBoxStyle::GetInstance(QStyle* style)
+{
+    std::lock_guard<std::mutex> lock(s_singleton_instances_mutex_);
+    auto inst_iter = s_singleton_instances_.constFind(style);
+    NoBigComboBoxStyle* inst = nullptr;
+    if(inst_iter == s_singleton_instances_.constEnd() || *inst_iter == nullptr) {
+        inst = new NoBigComboBoxStyle(style);
+        inst->setParent(APPLICATION);
+        s_singleton_instances_.insert(style, inst);
+        qDebug() << "QProxyStyle NoBigComboBox created for" << style->objectName() << style;
+    } else {
+        inst = *inst_iter;
+    }
+    return inst;
+}
+
```

Comment:
why did you put all of that out of line? makes sense to me to put everything here into `GetInstance`'s declaration, like:

```suggestion
    /** QProxyStyle and QStyle objects can't be free'd even if all the widgets using them are gone,
     * so make singletons tied to the lifetime of the application to clean them up and ensure they aren't 
     * being remade over and over again, thus leaking memory.
     */
    static NoBigComboBoxStyle* GetInstance(QStyle* style)
    {
        static QMap<QStyle*, NoBigComboBoxStyle*> s_singleton_instances_ = {};
        static std::mutex s_singleton_instances_mutex_;

        std::lock_guard<std::mutex> lock(s_singleton_instances_mutex_);
        auto inst_iter = s_singleton_instances_.constFind(style);
        NoBigComboBoxStyle* inst = nullptr;
        if(inst_iter == s_singleton_instances_.constEnd() || *inst_iter == nullptr) {
            inst = new NoBigComboBoxStyle(style);
            inst->setParent(APPLICATION);
            s_singleton_instances_.insert(style, inst);
            qDebug() << "QProxyStyle NoBigComboBox created for" << style->objectName() << style;
        } else {
            inst = *inst_iter;
        }
        return inst;
    }
};
```

Also, the `QMap` could be changed to a `QHash`, since it makes no sense to preserve ordering of pointer types (fix a clazy warning), and `GetInstance` could be changed to `getInstance`.

Lastly, I think with this singleton behavior, I think it makes sense to make the style's ctor private :slightly_smiling_face:

Reply:
Because non const statics must be initialized out of line per according to the compiler :stuck_out_tongue:

Reply:
what compiler? mine works fine with that :x

Reply:
clang

Reply:
mine is clang too... o.O

Reply:
![2023-05-28_04-53](https://github.com/PrismLauncher/PrismLauncher/assets/508861/ffd99bff-96ac-4bcc-92ec-bc4fc164d450)
and without the commented out lines I get link errors;

Reply:
Yeah, that's because you're declaring the static variables in the class, not inside the function scope. Then you really have to move those out of line. That's not what I suggested though :p

Reply:
but if they ... nvm. nothing outside the function needs access. and being static they won't be re initialized every run like I was initial thinking.  sometime C++ makes me overthink things. sometimes it's nuance escapes me.